### PR TITLE
WIP: Set up CI etc

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

**Comment by KristofferC** [comment]
The tests are so horribly made that they take forever to run so I think we need to run this on dedicated hardware and in parallel.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      - uses: julia-actions/setup-julia@v2
+        with:
+          version: 'nightly'
+      - name: Generate docs
+        run: |
+          julia --project --color=yes -e ''
+          julia --project --color=yes -e 'using Pkg; Pkg.respect_sysimage_versions(false); Pkg.activate("docs"); Pkg.develop(PackageSpec(path = pwd()))'
+          julia --project=docs --color=yes docs/make.jl
+        env:
+          DOCUMENTER_KEY: ${{ secrets.DOCUMENTER_KEY }}
```
Comment (fredrikekre) [inline .github/workflows/ci.yml:28]:
```suggestion
          DOCUMENTER_KEY: ${{ secrets.DOCUMENTER_KEY }}
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by ViralBShah** [comment]
Can we reduce the number of tests but without significantly losing coverage? Maybe a major exercise. I suppose the easiest thing is to do the excision with the same test suite (use buildkite the way we do with [LBT tests](https://github.com/JuliaLinearAlgebra/libblastrampoline/tree/main/.buildkite)) and then hopefully over time clean up the testsuite.

**Comment by KristofferC** [comment]
That's the plan, just throw our buildkite hardware resources at it and run it in parallel to not block this on anything.

**Pull request merged**